### DISCUSSION TITLE
CSV Import shows 0 page views

### DISCUSSION THREAD

**Discussion by thimiraonline** [plausible/analytics#4304] [Self-Hosted Support] [answered]
I'm seeking assistance with an issue I'm experiencing after importing page view data into my Plausible Analytics account.

Here are the details of the situation:

1. I successfully imported page view data using the CSV import method.
2. The import process seemed to complete without any apparent errors.
3. However, when I check my Plausible dashboard, it shows 0 page views for the imported data.

![Screenshot 2024-07-03 at 9 03 10 AM](https://github.com/plausible/analytics/assets/62184599/84a21a76-32ff-49eb-9363-292b59a8fd1c)

Specific details:
- Import method used: CSV import feature
- Plausible version: self-hosted version and tested on plausible.io also

Steps I've already taken:
1. Verified that the CSV format matches Plausible's requirements.
2. Checked that I'm looking at the correct date range in the dashboard.
3. Confirmed that the domain in the CSV matches my site configuration in Plausible.
5. Waited for several hours to allow for potential data aggregation delays.
6. Cleared browser cache and tried different browsers.

I would greatly appreciate your assistance in resolving this issue. Specifically, I'd like to know:

1. Was the data successfully imported on the server side?
2. If so, why isn't it appearing in my dashboard?
3. Are there any additional steps I need to take to make the imported data visible?

Imported CSV is here.

[imported_pages_20200411_20210330.csv](https://github.com/user-attachments/files/16076746/imported_pages_20200411_20210330.csv)

**Comment by ruslandoga** [comment, marked as answer]
👋 @thimiraonline 

The full import consists of nine CSV files. Please see https://plausible.io/docs/csv-import#csv-structure

> It's important to note that while all files and columns within those files are optional, to maintain completeness and avoid gaps in your dashboard, it's recommended that all files be present and fully populated according to the structures outlined here. 

A single `improted_pages` CSV is simply not enough (it populates `imported_pages` table which is mostly used for queries for "pages" section of the dashboard, e.g. https://plausible.io/plausible.io/pages). The UI shows 0 pageviews since the imported pageviews are stored in `imported_visitors` table and are read from [`imported_visitors.csv`](https://plausible.io/docs/csv-import#imported_visitors)

**Reply by thimiraonline** [reply]
@ruslandoga Thank you for replying quickly. 

Yes you are correct, I just checked with Postman with running API call. Pageviews are there. 

![Screenshot 2024-07-03 at 9 16 51 AM](https://github.com/plausible/analytics/assets/62184599/e756900f-3c7c-47af-8078-ea625b2d1913)


Should I have to prepare all csv mentioned in the file upload field?

**Reply by ruslandoga** [reply]
To get all sections of the dashboard working and showing data, yes.

**Reply by thimiraonline** [reply]
> To get all sections of the dashboard working and showing data, yes.

Can I import those files separately? My priority is page views and visits counts. Browser analytics and other things are in second place. My intention is to create a page view count for each page fro my Blog built with Ghost CMS. [https://github.com/TryGhost/Ghost](Ghost CMS) as they are not providing page views.

**Reply by ruslandoga** [reply]
You would need to remove the previous import (Delete Import) since it already occupies this time range, and create a new one with the CSV files that you want. For page views and visits counts and you probably need imported_pages and imported_visitors CSVs.

**Reply by thimiraonline** [reply]
> You would need to remove the previous import (Delete Import) since it already occupies this time range, and create a new one with the CSV files that you want. For page views and visits counts and you probably need imported_pages and imported_visitors CSVs.

I meant, can I import imported_pages 1st and then import imported_visitors later?

**Reply by ruslandoga** [reply]
No, since the first and second imports would try to occupy the same time range.

**Reply by thimiraonline** [reply]
@ruslandoga What does this mean? Once I make 5 imports, what will happen?

<img width="837" alt="Screenshot 2024-07-03 at 2 54 15 PM" src="https://github.com/plausible/analytics/assets/62184599/20200cd7-b0f9-4fff-b244-de30f029a663">

**Reply by ruslandoga** [reply]
It means that you would need to re-import CSV, I guess.

**Reply by thimiraonline** [reply]
Are there any limitation for file sizes?

**Reply by ruslandoga** [reply]
Not that I know of.

**Reply by thimiraonline** [reply]
@ruslandoga Thank you for your help. I couldn't have done this without you. Everything was successfully imported from Matomo to Plausible. There was an issue exporting Referrers from Matomo because we can only get data for a specific date through the API, not for a date range. I resolved this by creating a Node.js application that automated the data retrieval process.

![test androidwedakarayo com_androidwedakarayo com_period=all keybindHint=A](https://github.com/plausible/analytics/assets/62184599/61a689f5-2b25-44c5-a80a-ecd0ce56ccca)

**Comment by thimiraonline** [comment]
@ruslandoga issue resolved. Thank you for the help. I'll extract other details and prepare each csv files and import them. For now I imported pageviews and visitors.
